app/src/main/kotlin/com/exallium/todoapp/editnote/EditNotePresenter.kt

```diff
@@ -13,35 +14,48 @@ import timber.log.Timber
  * Presenter for Edit Note Screen
  */
 class EditNotePresenter(view: EditNoteView,
-                          private val model: EditNoteModel,
-                          private val screenBundleHelper: ScreenBundleHelper) : BasePresenter<EditNoteView>(view) {
+                        private val model: EditNoteModel,
+                        private val screenBundleHelper: ScreenBundleHelper,
+                        private val idFactory: IdFactory) : BasePresenter<EditNoteView>(view) {
 
-    private val showNewNoteDetailSubscriberFn: (Unit?) -> (Unit) = {
-        view.showNewNoteDetail(getArgs())
+    private val showAllNotesSubscriberFn: (Unit?) -> (Unit) = {
+        view.showAllNotes(getArgs())
     }
 
-    override fun onViewCreated() {
-        screenBundleHelper.setTitle(getArgs(), R.string.edit_note_screen_title)
-        val noteId: String = screenBundleHelper.getNoteId(getArgs())
-
-        setupGetNoteDetailSubscription(noteId)
-
-        setupSaveNoteSubscription(noteId)
+    private val showNoteDetailSubscriberFn: (Unit?) -> (Unit) = {
+        view.showNoteDetail(getArgs())
+    }
 
-        view.cancelEditNoteClicks().map { null }.subscribe(showNewNoteDetailSubscriberFn).addToComposite()
+    override fun onViewCreated() {
+        val args = getArgs()
+        if (screenBundleHelper.getNoteId(args) == null) {
```

Comment:
Let's refactor with some methods for these blocks, as they're quite large.  I feel like there's a really slick kotliny way to refactor this block, but lets start with the  intermediate functions.  Also, let's use an intermediary value for the note since you're accessing it again anyway in the bottom block.  The top block will check this intermediate value for null, and the bottom block will smart cast it to non-null for you.  It'll remove the need for !!.  Generally i find if you're utilizing !!, you might want to rethink the approach.